File: data/readable_format/cpp/file1137.cpp
#include <iostream>
#include <cassert>
#include <vector>
#include <algorithm> // For std::min and std::max

void clawpack46_rpt2adv_manifold(int ixy, int maxm, int meqn,
                                 int mwaves, int mbc, int mx,
                                 std::vector<std::vector<double>>& ql,
                                 std::vector<std::vector<double>>& qr,
                                 const std::vector<std::vector<double>>& aux1,
                                 const std::vector<std::vector<double>>& aux2,
                                 const std::vector<std::vector<double>>& aux3,
                                 int imp,
                                 std::vector<std::vector<double>>& asdq,
                                 std::vector<std::vector<double>>& bmasdq,
                                 std::vector<std::vector<double>>& bpasdq) {
    int iface = 3 - ixy;
    for (int i = 1 - mbc; i <= mx + mbc - 1; ++i) {
        int i1 = i - 2 + imp;
        for (int m = 0; m < meqn; ++m) {
            bmasdq[i + mbc][m] = std::min(aux2[i1 + mbc][iface], 0.0) * asdq[i + mbc][m];
            bpasdq[i + mbc][m] = std::max(aux3[i1 + mbc][iface], 0.0) * asdq[i + mbc][m];
        }
    }
}

int main() {
    const int maxm = 5, meqn = 2, mwaves = 3, mbc = 1, mx = 3, imp = 1;
    const int ixy = 1;
    
    std::vector<std::vector<double>> ql((maxm + 2*mbc), std::vector<double>(meqn));
    std::vector<std::vector<double>> qr((maxm + 2*mbc), std::vector<double>(meqn));
    std::vector<std::vector<double>> aux1((maxm + 2*mbc), std::vector<double>(mwaves));
    std::vector<std::vector<double>> aux2((maxm + 2*mbc), std::vector<double>(mwaves, 0.5)); // Initialized to ensure min/max logic
    std::vector<std::vector<double>> aux3((maxm + 2*mbc), std::vector<double>(mwaves, 0.5)); // Initialized to ensure min/max logic
    std::vector<std::vector<double>> asdq((maxm + 2*mbc), std::vector<double>(meqn, 1.0)); // Initialized to 1.0 for simplicity
    std::vector<std::vector<double>> bmasdq((maxm + 2*mbc), std::vector<double>(meqn));
    std::vector<std::vector<double>> bpasdq((maxm + 2*mbc), std::vector<double>(meqn));

    // Call the translated function
    clawpack46_rpt2adv_manifold(ixy, maxm, meqn, mwaves, mbc, mx, ql, qr, aux1, aux2, aux3, imp, asdq, bmasdq, bpasdq);

    // Simple checks (assuming specific behavior based on the initial setup)
    for (int i = 0; i < maxm + 2*mbc; ++i) {
        for (int m = 0; m < meqn; ++m) {
            assert(bmasdq[i][m] <= 0.0); // bmasdq should be <= 0 due to min logic with aux2 initialized to 0.5
            assert(bpasdq[i][m] >= 0.0); // bpasdq should be >= 0 due to max logic with aux3 initialized to 0.5
        }
    }

    std::cout << "All tests passed." << std::endl;

    return 0;
}

File: data/readable_format/fortran/file1137.f90
program test_clawpack46
    implicit none

    ! Declaration of variables
    integer, parameter :: maxm = 5, meqn = 2, mwaves = 3, mbc = 1, mx = 3, imp = 1
    integer :: ixy
    double precision :: ql(1-mbc:maxm+mbc, meqn)
    double precision :: qr(1-mbc:maxm+mbc, meqn)
    double precision :: aux1(1-mbc:maxm+mbc, mwaves)
    double precision :: aux2(1-mbc:maxm+mbc, mwaves)
    double precision :: aux3(1-mbc:maxm+mbc, mwaves)
    double precision :: asdq(1-mbc:maxm+mbc, meqn)
    double precision :: bmasdq(1-mbc:maxm+mbc, meqn)
    double precision :: bpasdq(1-mbc:maxm+mbc, meqn)

    ! Initialize some variables with arbitrary values for testing
    ixy = 2
    call random_number(ql)
    call random_number(qr)
    call random_number(aux1)
    call random_number(aux2)
    call random_number(aux3)
    call random_number(asdq)

    ! Calling the subroutine
    call clawpack46_rpt2adv_manifold(ixy, maxm, meqn, mwaves, mbc, mx, ql, qr, aux1, aux2, aux3, imp, asdq, bmasdq, bpasdq)

    ! Optionally: Output values or perform checks

contains

    subroutine clawpack46_rpt2adv_manifold(ixy,maxm,meqn,mwaves,mbc,mx,ql,qr,aux1,aux2,aux3,imp,asdq,bmasdq,bpasdq)
        implicit none

        integer :: ixy,maxm, meqn, mwaves, mbc, mx, imp, m
        double precision, dimension(1-mbc:maxm+mbc, meqn) :: ql, qr, asdq, bmasdq, bpasdq
        double precision, dimension(1-mbc:maxm+mbc, *) :: aux1, aux2, aux3

        integer :: iface, i, i1

        iface = 3-ixy  
        do i = 2-mbc,mx+mbc
            i1 = i-2+imp    
            do m = 1,meqn
                bmasdq(i,m) = min(aux2(i1,iface+1), 0.d0) * asdq(i,m)
                bpasdq(i,m) = max(aux3(i1,iface+1), 0.d0) * asdq(i,m)
            end do
        end do

    end subroutine clawpack46_rpt2adv_manifold

end program test_clawpack46